File: Python/560. Subarray Sum Equals K.py
class Solution:
    def subarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        maple = dict()
        res, ans = 0, 0
        maple[0] = 1;
        for num in nums:
            res += num
            ans += maple.get(res-k, 0)
            maple[res] = maple.get(res, 0) + 1
        return ans

File: C++/560. Subarray Sum Equals K.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> maple;
        maple[0] = 1;
        int sum = 0, ans = 0;
        for(auto num: nums)
        {
            sum += num;
            if(maple.find(sum - k) != maple.end())
                ans += maple[sum-k];
            maple[sum] += 1;
        }
        return ans;
    }
};